pkg/env/versionsetting.go

```diff
@@ -0,0 +1,62 @@
+package env
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/Masterminds/semver"
+	"github.com/stackrox/rox/pkg/utils/panic"
+)
+
+// VersionSetting represents an environment variable which should be parsed into a semver version
+type VersionSetting struct {
+	envVar       string
+	defaultValue *semver.Version
+}
+
+// EnvVar returns the string name of the environment variable
+func (s *VersionSetting) EnvVar() string {
+	return s.envVar
+}
+
+// DefaultValue returns the default value for the setting
+func (s *VersionSetting) DefaultValue() *semver.Version {
+	return s.defaultValue
+}
+
+// Setting returns the string form of the version environment variable
+func (s *VersionSetting) Setting() string {
+	return fmt.Sprintf("%s", s.VersionSetting().String())
+}
+
+// VersionSetting returns the semver.Version object represented by the environment variable
+func (s *VersionSetting) VersionSetting() *semver.Version {
+	val := os.Getenv(s.envVar)
+	if val == "" {
+		return s.defaultValue
+	}
+
+	version, err := semver.NewVersion(val)
+	if err != nil {
+		return s.defaultValue
+	}
+
+	return version
+}
+
+// RegisterVersionSetting globally registers and returns a new version setting.
+func RegisterVersionSetting(envVar string, defaultValue string) *VersionSetting {
```

Comment:
How would the support look like with this env var? By allowing the version to be configured we might run into problems with customers setting this to something lower than 1.6.0 to, for example, change the CO's heath status. If they do that, most of the functionality won't work. We should make clear that this is for internal uses only and that we do not support < 1.6.0.

Reply:
Maybe it would make sense to have a hardcoded minimum version and not allow this variable to put anything lower than that. That way we can configure a higher version if needed but never lower. WDYT?

Reply:
Luis, I argued that if customers set it to wrong value then they are actually breaking the feat on purpose. I voted to add this as env var to give us (and support engs) a tool to fix unexpected problems without waiting for a release.

I think that not allowing anything that is lower than 1.6.0 as a general rule would be also okay.

Reply:
> I voted to add this as env var to give us (and support engs) a tool to fix unexpected problems without waiting for a release

I'm totally fine with keeping the env var. My concern was more towards not allowing anything lower than 1.6.0.

> I think that not allowing anything that is lower than 1.6.0 as a general rule would be perfectly fine.

Yeah I think we should have that limit. We know for sure we are not going to work properly in versions < 1.6.0.

Reply:
I have added support for the minimal version to the version settings.